infra/terraform/modules/aws_account_logging/cloudtrail/cloudtrail_to_es.py

```diff
@@ -47,28 +49,41 @@ def post_to_elasticsearch(doc):
 
     # TODO handle errors
     response = requests.post(
-        elasticsearch_url(os.environ),
+        elasticsearch_url(os.environ, get_pipeline(doc)),
         auth=elasticsearch_auth_header(os.environ),
         json=doc)
 
     LOG.debug('Elasticsearch response: {status} - {text}'.format(
         status=response.status_code,
         text=response.text))
 
+def get_pipeline(doc):
+    # sourceIPAddress may not be present, or may contain a hostname,
+    # which causes an ElasticSearch error - so only specify the geoip
+    # pipeline if an actual IP address is present
+    try:
+        ipaddress.ip_address(doc['sourceIPAddress'])
+        return 'cloudtrail-geoip'
+    except (KeyError, ValueError):
+        return None
```

Comment:
It's better to minimize the code inside a `try` block. You can also avoid the later check by returning an empty string instead of `None` if there is an exception. And if you use `doc.get()` you only need to catch `ValueError`:
```python
try:
    ipaddress.ip_address(doc.get('sourceIPAddress'))

except ValueError:
    return 'pipeline=cloudtrail-geoip'

return ''
```